### PR TITLE
Remove new platform members from OSPlatform

### PR DESCRIPTION
**Pull request by adamsitnik**
The spec mentions removing also the "Browser": https://github.com/dotnet/runtime/issues/40111#issuecomment-668771421

I am not sure about this since it was added 4 months ago together with `Architecture.Wasm` in https://github.com/dotnet/runtime/pull/34781

cc @terrajobst

### CODE REVIEW

**Comment by adamsitnik** [comment]
I've added the no merge label as I simply need to revert more changes (new methods added to RuntimeInformation && OrdinalIgnoreCaseChange)

**Comment by jeffhandley** [comment]
@terrajobst / @adamsitnik -- I propose that we leave `OSPlatform.Browser` in place for this PR since there are already some references to it in some of our tests. We'd then keep this PR to reverting @adamsitnik's PR that had added the other members and changed the string comparison.

Here are a couple of PRs that had added references to `OSPlatform.Browser`:

1. PR #39130
1. PR #39355

I was considering a suggestion that we change the reference to `OSPlatform.Browser` to use `OperatingSystem.IsBrowser()`, but the references to `OSPlatform.Browser` in those files is alongside other `OSPlatform` references, which would make the work snowball.

**Comment by steveisok** [comment]
I would definitely feel more comfortable if we left `OSPlatform.Browser` in place.  Snowballs at this point in time would be pretty bad.

**Comment by akoeplinger** [comment]
I don't think it's that bad, there are only two usages of `OSPlatform.Browser` outside of the RuntimeInformation tests. We can trivially change those.

**Comment by campersau** [comment]
Note that aspnetcore is also already using `OSPlatform.Browser`: https://github.com/search?l=C%23&p=1&q=%22osplatform.browser%22&type=Code

**Comment by akoeplinger** [comment]
Yeah we can change them back to using `RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER"))`. I'd still prefer that to releasing with a new property that we immediately obsolete.

**Comment by adamsitnik** [comment]
@buyaa-n @jeffhandley PTAL. I've removed the methods and properties that I've recently added.

Once #40457 gets merged I am going to send another PR that will make `RuntimeInfromation.IsOSPlatform` just call `OperatingSystem.IsOSPlatform`, remove `Browser` and use new `OperatingSystem` methods everywhere in runtime and aspnet

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs
Comment (eerhardt) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:21]: I think `Browser` should be removed as well, since we added it in `net5.0`

**Review by jeffhandley — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs
Comment (jeffhandley) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:21]: @adamsitnik mentioned he'll send another PR for removing `Browser` once #40457 gets merged.

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Comment by lewing** [comment]
You'll need to fix dotnet/aspnetcore as well https://github.com/dotnet/aspnetcore/search?q=OSPlatform.Browser&unscoped_q=OSPlatform.Browser

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs
Comment (adamsitnik) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:21]: @eerhardt I would prefer to do it in the next PR where I will replace all usages of `RuntimeInfromation.IsOSPlatform(XYZ)` with  `OperatingSystem.IsXYZ` new method calls

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs
Comment (eerhardt) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs:20]: Do we still need this change we made in 5.0 now that we are backing out the other changes?

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs
Comment (eerhardt) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:21]: Ok - I'm not sure I understand the difference between `Browser` and `iOS` in this area, but that is fine with me.

---
In reply to: [466574218](https://github.com/dotnet/runtime/pull/40373#discussion_r466574218) [](ancestors = 466574218)

Comment (jeffhandley) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:21]: `Browser` was added earlier in 5.0 and has consumers; `iOS` and these others were just added in Preview 8 and doesn't yet have consumers.

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs
Comment (jeffhandley) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs:20]: Good catch. OK with me to remove it.

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs
Comment (adamsitnik) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs]: ok, I've removed browser and replace it's usages:

```diff
-RuntimeInformation.IsOSPlatform(OSPlatform.Browser)
+OperatingSystem.IsBrowser()
```

since #40457 has not been merged yet, the CI is going to go red.

Once #40457 CI is green and merged I am going to sync this PR, wait until it goes green and merge it. And then send a PR to ASP.NET repo

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/tests/CheckPlatformTests.cs
Comment (adamsitnik) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/tests/CheckPlatformTests.cs:56]:
```suggestion
```

**Comment by adamsitnik** [comment]
The CI failure is unrelated (`Utf8String` long running test timeout), merging

**Comment by adamsitnik** [comment]
I've sent a PR to ASP.NET: https://github.com/dotnet/aspnetcore/pull/24652

/cc @jeffhandley @eerhardt

**Pull request merged**
